assets/js/components/product-attribute-control/index.js

```diff
@@ -2,112 +2,45 @@
  * External dependencies
  */
 import { __, _n, sprintf } from '@wordpress/i18n';
-import { addQueryArgs } from '@wordpress/url';
-import apiFetch from '@wordpress/api-fetch';
 import { Component, Fragment } from '@wordpress/element';
-import { debounce, find } from 'lodash';
+import { find } from 'lodash';
 import PropTypes from 'prop-types';
 import { SearchListControl, SearchListItem } from '@woocommerce/components';
 import { SelectControl, Spinner } from '@wordpress/components';
-import { ENDPOINTS } from '@woocommerce/block-settings';
 
 /**
  * Internal dependencies
  */
+import { withAttributes } from '../../hocs';
+import ErrorMessage from '../api-error-placeholder/error-message.js';
 import './style.scss';
 
 class ProductAttributeControl extends Component {
 	constructor() {
-		super( ...arguments );
-		this.state = {
-			list: [],
-			loading: true,
-			attribute: 0,
-			termsList: {},
-			termsLoading: true,
-		};
+		super();
 
-		this.debouncedGetTerms = debounce( this.getTerms.bind( this ), 200 );
 		this.renderItem = this.renderItem.bind( this );
```

Comment:
The removal of lifecycle methods and most of the constructor seems to make this a good candidate for a functional component now?

Reply:
Done: 31b8cd2d56c867058ab1ed60304d4a15f59fa50c.